scripts/fn_paradrop.sqf

```diff
@@ -0,0 +1,216 @@
+    _dropData = DROP_MODES select _dropMode;
+} else {
+    if (_dropMode isEqualType [] && {count _dropMode == 3}) then {
+        _dropData = _dropMode;
+    };
+};
+
+_dropData params ["_dropHeight", "_dropLength", "_chuteHeight"];
+
+private _vehicle = vehicle (leader _pilotGroup);
+if (isNull _vehicle) exitWith {ERROR("Paradrop pilot group not in vehicle!")};
+
+private _origHeight = (getPosATL _vehicle) select 2;
+_vehicle flyInHeight _dropHeight; // set forced flying height
+if (_resetHeight) then {
+    [{
+        params ["_vehicle"];
+        (fullCrew [_vehicle,"cargo"]) isEqualTo []
+    }, {
+        params ["_vehicle", "_origHeight"];
+        _vehicle flyInHeight _origHeight;
+    }, [_vehicle, _origHeight]] call CBA_fnc_waitUntilAndExecute;
+};
+
+// Use addWayPoint for the group piloting the vehicle to add the drop waypoints
+private _dropRunOrigin = getPosATL _vehicle;
+
+// Obtain a random location within drop target
+// Between current position and drop target, find angle
+private _dropTarget = [_loc, true] call CBA_fnc_randPosArea;
+
+if (_dropTarget isEqualTo []) exitWith {ERROR("Paradrop target for marker '%1' not found!",_loc)};
+
+private _fnc_dzLocs = {
+    params ["_dropTarget", "_dropRunOrigin", "_dropLength"];
+    // getDir gives the azimuth, which differs in convention from trig
+    // in two ways: (1) azimuth has zero origin at North and
+    // (2) advances clockwise rather than counterclockwise
+    // To convert, use a conditional function:    
+    private _azimuth = _dropRunOrigin getDir _dropTarget;
+    private _theta = switch (floor _azimuth/90) do {
+        case 0: {90-_azimuth;};
+        default {450-_azimuth;};
+    };
+    // For a known drop point and known radius with trig angle _theta,
+    // and radius r given by _dropLength/2,
+    // then coordinates in the xy-plane relative to _dropTarget are:
+    // +/- r*cos _theta for x (Northing) and +/- r*sin _theta for x (Easting)
+
+    private _yDiff = (sin _theta) * _dropLength/2;
+    private _xDiff = (cos _theta) * _dropLength/2;
+    // set all coordinates based on DZ points in xy-plane; z coord does not matter
+    // double lengths on second DZ coordinate to ensure straight line flight
+    _dropTarget params ["_dropTargetX", "_dropTargetY"];
+    [
+        [_dropTargetX + _xDiff, _dropTargetY + _yDiff, _dropHeight],
+        [_dropTargetX - _xDiff, _dropTargetY - _yDiff, _dropHeight]
+    ];
+};
+
+private _dropZoneLocs = [_dropTarget, _dropRunOrigin, _dropLength] call _fnc_dzLocs;
+private _minDistance = _dropRunOrigin distance (_dropZoneLocs select 1);
+
+// Handling for when calling waypoint is within distance _dropLength
+if (_minDistance < _dropLength) then {
+    _dropLength = _minDistance - 100; //arbitrary distance to allow for a helicopter turn
+    if (_dropLength > 0) then {
+        _dropZoneLocs = [_dropTarget,_dropRunOrigin, _dropLength] call _fnc_dzLocs;
+    } else {
+        diag_log("Warning: Vehicle too close to DZ!");
```

Comment:
I am still not sure if being too close should _exit/abort_ the drop or just continue with the location found on L142?

If it should abort it, you should write it like so:
```sqf
private _withinDistance = _minDistance < _dropLength;
private _dropLength = _minDistance - 100; // arbitrary distance to allow for a helicopter turn
if (_withinDistance && {_dropLength < 0}) exitWith {
    ERROR_1("Paradrop vehicle '%1' too close to DZ!",_vehicle);
};
if (_withinDistance) then {
    _dropZoneLocs = [_dropTarget,_dropRunOrigin, _dropLength] call _fnc_dzLocs;
};
```
or like so:
```sqf
private _tooClose = false;
if (_minDistance < _dropLength) then {
    _dropLength = _minDistance - 100; //arbitrary distance to allow for a helicopter turn
    if (_dropLength < 0) exitWith {
        _tooClose = true;
    };
    _dropZoneLocs = [_dropTarget,_dropRunOrigin, _dropLength] call _fnc_dzLocs;
}; 

if (_tooClose) exitWith {ERROR_1("Paradrop vehicle '%1' too close to DZ!",_vehicle)};
```

If it should just continue with the earlier location, write this log like so and leave the rest of it as it is:
```sqf
WARNING_1("Paradrop vehicle '%1' too close to DZ!",_vehicle);
```

Reply:
Took final suggestion.